### PR TITLE
add rocksdb log dir option

### PR DESCRIPTION
**Pull request by overvenus**
Add a ffi `rocksdb_options_set_db_log_dir`

PTAL @siddontang @BusyJay @zhangjinpeng1987

cc pingcap/tikv#1360

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by overvenus** [comment]
cc pingcap/tikv#1391

File: src/rocksdb.rs [hunk 1]
```diff
@@ -1218,6 +1219,33 @@ mod test {
             assert!(r.unwrap().unwrap().to_utf8().unwrap() == str::from_utf8(value).unwrap());
         }
     }
+
+    #[test]
+    fn log_dir_test() {
+        let db_dir = TempDir::new("_rust_rocksdb_logdirtest").unwrap();
+        let db_path = db_dir.path().to_str().unwrap();
+        let mut log_path = db_path.to_owned();
```
Comment (BusyJay) [inline src/rocksdb.rs]: Use `join` instead.

File: src/rocksdb_options.rs [hunk 2]
```diff
@@ -543,6 +543,13 @@ impl Options {
             rocksdb_ffi::rocksdb_options_set_num_levels(self.inner, n);
         }
     }
+
+    pub fn set_db_log_dir(&mut self, path: &str) {
```
Comment (BusyJay) [inline src/rocksdb_options.rs]: tests?

Comment (overvenus) [inline src/rocksdb_options.rs]: See [line 1224](https://github.com/ngaut/rust-rocksdb/pull/61/files/7ceb90917844a2fa178e9a511db9f9505616a5cf#diff-15fa9749208206c5a07bbe0cc9aa11abR1224).

Comment (BusyJay) [inline src/rocksdb_options.rs]: But these few lines are not covered.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**